sections/transactions.js

```diff
@@ -102,5 +102,30 @@ export default [
       "Calling `trx.rollback` will return a rejected Promise. If you don't pass any argument to `trx.rollback`, a generic `Error` object will be created and passed in to ensure the Promise always rejects with something.",
       "Note that Amazon Redshift does not support savepoints in transactions.",
     ]
+  },
+  {
+    type: "text",
+    content: "In some cases you may prefer to create transaction but only execute statements in it later. In such case call method `transaction` without a handler function:"
+  },
+  {
+    type: "code",
+    language: "js",
+    content: `
+      // Using trx as a transaction object:
+      const trx = await knex.transaction();
+
+      knex.insert({name: 'Old Books'}, 'id')
```

Comment:
I would prefer syntax:

```
trx('catalogues')
  .insert({name: 'Old Books'}, 'id')
  .then(function(ids) {
    books.forEach(book =>  book.catalogue_id = ids[0]);
    return trx('books').insert(books);
  })
  .then(trx.commit)
  .then(trx.rollback);

```

How its written currently using `.transacting(trx)` is not recommended, because it is easy to forget from some query and hard to spot from code and it is spamming unnecessary amount of queries to server.

We have too much  antique example code in docs :)

Reply:
Updated both original example and this one, please check if I didn't mess it up :D